litestar/middleware/base.py

```diff
@@ -249,3 +255,38 @@ async def handle(self, scope: Scope, receive: Receive, send: Send, next_app: ASG
         """
 
         raise NotImplementedError
+
+
+class ASGIAuthenticationMiddleware(ASGIMiddleware):
```

Comment:
1. Shouldn't this go into `middleware.authentication`?
2. Maybe rename to `BaseAuthenticationMiddleware`?